### ISSUE TITLE
Maxpool fails in Model_net

### ISSUE THREAD

**Issue by HariniMohan0102** [tenstorrent/tt-metal#11160]
**Describe the bug**
Here is the unit tests of failing maxpool op of each resolution:

**When Maxpooling=True, Encoder res: 4094x510**
- To reproduce the issue, run the command: `pytest tests/ttnn/unit_tests/operations/test_maxpool2d.py::test_model_net_max_pool_4094x510`
- Among 4 maxpools, 2 passed and 2 maxpools fails with OOM issue.

**When Maxpooling=True, Encoder res: 2047x255**
- To reproduce the issue, run the command: `pytest tests/ttnn/unit_tests/operations/test_maxpool2d.py::test_model_net_max_pool_2047x255`
- Among 4 maxpools, 3 maxpools passed, 1 maxpools fails with OOM issue.

**To Reproduce**
1. Checkout to the branch [harini/model_net_failing_maxpools](https://github.com/tenstorrent/tt-metal/compare/harini/model_net_failing_maxpools)
2. Run the respective commands of Ops in each input resolution to reproduce the issues.

**Expected behavior**
To run the op for the specific input configurations without error.

**Please complete the following environment information:**
- OS: [e.g. Ubuntu 20.04]
- Device: n150

**Comment by sankarmanoj-tt** [comment]
Maxpool supports only bfloat16. So any bfloat8 inputs must be converted to bfloat16. My guess is that when this conversion happens, there isn't enough memory to fit both input and output tensors.

**Comment by mywoodstock** [comment]
Yeah, basically you can provide bfp8_b input to the maxpool op. Internally the halo op converts to RM, so BFP16. If input is already BFP16, it should need less memory.

**Comment by mywoodstock** [comment]
@HariniMohan0102 Could you please retry the tests with the latest `main`? A few fixes went in, which have resolved the page size errors. The OOM ones may still fail. Please let me know what you see. Thanks!

**Comment by HariniMohan0102** [comment]
@mywoodstock  I'll test with latest main and update soon.

**Comment by HariniMohan0102** [comment]
@mywoodstock I'll tested with latest main, cases with valid page size errors are resolved. 
3 cases fails with OOM error. Updated them to the current branch, please check.
Thank you.

**Comment by mywoodstock** [comment]
The 3 OOM cases are basically due to the large tensor shape, which comes out to requiring over 2MB per shard, which is beyond the L1 size. We do not have a solution for this currently, but we have it in the roadmap to make large sizes work through DRAM slicing. cc: @mbahnasTT

**Comment by shwetankTT** [comment]
@mbahnasTT Is it blocking for you? One software solution is to slice the tensor into smaller sub-tensors along the channel dimension, making each slice more manageable for the cache. We can then apply the max-pooling operation individually to each sub-tensor and concatenate the results afterward to reconstruct the final output.

**Comment by mbahnasTT** [comment]
@mywoodstock @shwetankTT OK with the workaround, and we'll update them on the current limitations and the roadmap plans.

**Comment by mbahnasTT** [comment]
@mywoodstock I downgraded it to P1 not P0, please treat it accordingly.

**Comment by dvartaniansTT** [comment]
@esmalTT since you are working on this model now. wanted to keep you in the loop regarding this issue. Thanks all.

**Comment by dvartaniansTT** [comment]
@HariniMohan0102 your branch has only 1 test for `2047x255 maxpool` but in issue description you are describing 4 of them. 
I can only see 1. I rebased to main and still that single one fails with OOM. cc: @esmalTT and @mbahnasTT 

@HariniMohan0102 pls double check the pytest on your branch and make sure the unit tests are all inclusive.

**Comment by HariniMohan0102** [comment]
@dvartaniansTT, I have the issue description is of when the issue was created. Recently, on rebasing to latest main, some of the cases passed, and few fails. I have updated the branch and readme previously. I'll update the issue description.

**Comment by mbahnasTT** [comment]
@rjakovljevicTT I'll close this ticket, as your team owns it now, and we'll revisit the issue if needed
